client/my-sites/themes/theme-showcase.jsx

```diff
@@ -48,15 +49,41 @@ const ThemeShowcase = React.createClass( {
 		this.setState( { showPreview: ! this.state.showPreview, previewingTheme: theme } );
 	},
 
-	onPreviewButtonClick( theme ) {
-		const { defaultOption } = this.props;
+	onPrimaryPreviewButtonClick( theme ) {
+		const option = this.getPrimaryOption();
 		this.setState( { showPreview: false }, () => {
-			defaultOption.action && defaultOption.action( theme );
+			option.action && option.action( theme );
 		} );
 	},
 
+	onSecondaryPreviewButtonClick( theme ) {
+		const { secondaryOption } = this.props;
+		this.setState( { showPreview: false }, () => {
+			secondaryOption && secondaryOption.action ? secondaryOption.action( theme ) : null;
+		} );
+	},
+
+	getPrimaryOption() {
```

Comment:
The more I work on this, though, I think that should be a separate PR, because IMO:

- There are a lot of variables that go into that decision (all the `options`, `isLoggedIn`, `isActive`, `price`) and something like `getPrimaryThemeAction()` is maybe not the best construct for this. 
- It expands this PR's effect to the primary actions in the theme sheet, making for even more needed testing.
- By itself, it would be a relatively small code change, which means easier review, testing, etc.